nixos/modules/services/desktops/pipewire/pipewire.nix

```diff
@@ -72,15 +73,103 @@ in {
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Filter-Chain
+          '';
+        };
+        client = mkOption {
+          type = lib.types.attrsOf json.type;
+          default = {};
+          example = {
+            "10-no-resample" = {
+              "stream.properties" = {
+                "resample.disable" = true;
+              };
+            };
+          };
+          description = lib.mdDoc ''
+            Configuration for PipeWire Client. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-client
+          '';
+        };
+        client-rt = mkOption {
+          type = lib.types.attrsOf json.type;
+          default = {};
+          example = {
+            "10-alsa-linear-volume" = {
+              "alsa.properties" = {
+                "alsa.volume-method" = "linear";
+              };
+            };
+          };
+          description = lib.mdDoc ''
+            Configuration for PipeWire Client-rt. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-client and
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-ALSA
+          '';
+        };
+        jack = mkOption {
+          type = lib.types.attrsOf json.type;
+          default = {};
+          example = {
+            "20-hide-midi" = {
+              "jack.properties" = {
+                "jack.show-midi" = false;
+              };
+            };
+          };
+          description = lib.mdDoc ''
+            Configuration for PipeWire Jack. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-JACK
+          '';
+        };
+        pipewire-pulse = mkOption {
+          type = lib.types.attrsOf json.type;
+          default = {};
+          example = {
+            "15-force-s16-info" = {
+              "pulse.rules" = [{
+                matches = [
+                  { "application.process.binary" = "teams"; }
+                  { "application.process.binary" = "teams-insiders"; }
+                  { "application.process.binary" = "skypeforlinux"; }
+                ];
+                actions = {
+                  quirks = [ "force-s16-info" ];
+                };
+              }];
+            };
+          };
+          description = lib.mdDoc ''
+            Configuration for PipeWire Pulse. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-PulseAudio and
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Filter-Chain
+          '';
+        };
+      };
     };
   };
 
   imports = [
-    (lib.mkRemovedOptionModule ["services" "pipewire" "config"] ''
-      Overriding default Pipewire configuration through NixOS options never worked correctly and is no longer supported.
-      Please create drop-in files in /etc/pipewire/pipewire.conf.d/ to make the desired setting changes instead.
```

Comment:
Maybe we should rename the new option to something like `extraConfig`, and then update this assert to point people to it?

Reply:
renamed the option to extraConfig and updated the assert to tell people to use the new methode